src/middleware/builtin.ts

```diff
@@ -210,30 +211,40 @@ export function matchConstraints(
  * Middleware that filters out messages that don't match pattern
  */
 export function matchMessage(
-  pattern: string | RegExp,
+  pattern: string | RegExp | (string | RegExp)[],
 ): Middleware<SlackEventMiddlewareArgs<'message' | 'app_mention'>> {
   return async ({ event, context, next }) => {
-    let tempMatches: RegExpMatchArray | null;
-
+    let tempMatches: RegExpMatchArray | null = null;
+    let patternArray: (string | RegExp)[] = [];
     if (!('text' in event) || event.text === undefined) {
       return;
     }
-
-    // Filter out messages or app mentions that don't contain the pattern
-    if (typeof pattern === 'string') {
-      if (!event.text.includes(pattern)) {
+    // Filter out messages or app mentions that don't contain the pattern]
+    if (typeof pattern === 'string' || util.types.isRegExp(pattern)) {
+      patternArray = [pattern];
+    } else {
+      patternArray = pattern;
+    }
+    let matched = false;
+    patternArray.forEach((patternArrayItem) => {
```

Comment:
What happens if multiple patterns match with `event.text`? I think we can stop checking other patterns, and then immediately call `next()` (and set `context["matches"]`) in that case. Also, it'd be great to have a test code covering this scenario.

@mwbrooks @misscoded @stevengill Do you think my suggestion here is valid?

Reply:
Yeah, we had a question about that scenario.  What if you have an array like this ["abc", /def/, /gh/] Assume the string and the 2 regex's all match.  In this case you would want context["matches"] to be empty since the string was the first to match?

Reply:
>What happens if multiple patterns match with event.text? I think we can stop checking other patterns, and then immediately call next() (and set context["matches"]) in that case.

I agree. I'd expect matching an array of patterns to work like an `OR` conditional. The first match would trigger an exit and set `context['matches']`.

>Yeah, we had a question about that scenario. What if you have an array like this ["abc", /def/, /gh/] Assume the string and the 2 regex's all match. In this case you would want context["matches"] to be empty since the string was the first to match?

Thanks for chiming in! Following the current logic, it would make sense that `context['matches']` would be empty if the `string` was the first match. However, is this a good experience for the client-developer code since you can't rely on `context['matches']` to always exist? (It may also be out-of-scope of this PR)

Reply:
>I'd expect matching an array of patterns to work like an OR conditional. The first match would trigger an exit and set context['matches'].

I agree. I think this is a straight forward implementation, too.

>However, is this a good experience for the client-developer code since you can't rely on context['matches'] to always exist? (It may also be out-of-scope of this PR)

We may want to have a warning log saying "You've mixed strings and regular expressions in app.message listener. With this settings, the existence of `context["matches"]` is not always guaranteed."  or something similar. But this is totally optional. We can revisit this if we receive many questions about this behavior.